lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -490,6 +491,39 @@ FEA::FunctionEntrypointAnalyzer(llvm::Module &M,
   initializeFakeFunctions();
 }
 
+static void serializeFunctionMetadata(llvm::LLVMContext &Context,
+                                      GeneratedCodeBasicInfo &GCBI,
+                                      const FunctionAnalysisResults &Properties,
+                                      const TupleTree<model::Binary> &Binary) {
+  using namespace llvm;
+
+  for (const auto &Function : Binary->Functions) {
+    if (Function.Type == FunctionTypeValue::Fake)
+      continue;
+
+    auto &CFG = Properties.at(Function.Entry).CFG;
+    // Already serialized?
+    if (CFG.size() == 0)
+      continue;
```

Comment:
How is this possible?

Reply:
Uhmm, I think that's not exactly what I wanted to achieve here. I drop it, but now that I think about it, does it make sense to serialize function metadata only if `Function.Type != FunctionTypeValue::Fake` and  `Function.Type != FunctionTypeValue::Invalid`? If for some reason is `Invalid`, we should not want to serialize the CFG, should we?